tests/unit/test_main.py

```diff
@@ -81,41 +81,37 @@ def load_plugins_mock(mocker, monkeypatch):
     return load_plugins_mock
 
 
-args = 'error_type'
-params = [
-    'list', 'obj'
-]
-
-
-@pytest.mark.parametrize(args, params)
-def test_log_or_exit_on_exceptions_no_debug(error_type, plugin_exc_mock,
-                                            mocker, monkeypatch):
+@pytest.mark.parametrize('error_type', ['list', 'obj'])
+def test_log_or_exit_on_exceptions_no_debug(
+        error_type, plugin_exc_mock, caplog, capsys):
     """Raise SystemExit if debug flag is off."""
-    logging_mock = mocker.MagicMock(main.logging, autospec=True)
-    monkeypatch.setattr(main, 'logging', logging_mock)
-
-    error = ('bad.plugin', plugin_exc_mock)
+    error = plugin_exc_mock
     if error_type == 'list':
-        error = [('bad.plugin', error)]
+        error = [error]
     with pytest.raises(SystemExit) as e:
         main._log_or_exit_on_exceptions('base msg', error, debug=False)
-
     e.match('1')
-    logging_mock.error.assert_called_once()
-    logging_mock.warn.assert_not_called()
 
+    assert 1 == len(caplog.records)
+    for record in caplog.records:
+        assert 'ERROR' == record.levelname
 
-def test_log_or_exit_on_exceptions_debug(plugin_exc_mock, mocker, monkeypatch):
-    """Do not exit out if debug flag is on."""
-    logging_mock = mocker.MagicMock(main.logging, autospec=True)
-    monkeypatch.setattr(main, 'logging', logging_mock)
+    out, err = capsys.readouterr()
+    assert '--- Logging error ---' not in err
```

Comment:
Uh, what's this for?

Reply:
The change I made is to actually log the error, rather than raise another error while trying to log the error.  If you check the example output I put in the PR description, you'll see that this was in the output previous to the change, and I am now verifying that it is no longer present.  

For context, I changed this test in two steps. First, I replaced the logging mock with capsys, because using a mock meant that the logging error was hidden and never triggered in tests. Once I was sure that it would be triggered, I added this line to make the test fail until I changed the buggy code to be sure I fixed the underlying issue once the test passed.

Reply:
I guess I'm not clear on why it was printed originally.  I also feel like this is going to need more context for someone reading this code cold.

Reply:
It was printed originally because we were passing the incorrect parameters to [logging.error](https://docs.python.org/3.6/library/logging.html#logging.Logger.error), which expects an exc_info object as the exc_info argument, not a tuple of plugin name and exc_info.

Reply:
The purpose of adding this line to the test is that once a bug occurs and is discovered, it should never happen again. I have added a line referring to the PR.